pkg/kubelet/prober/results/results_manager.go

```diff
@@ -87,3 +115,26 @@ func (m *manager) Remove(id kubecontainer.ContainerID) {
 	defer m.Unlock()
 	delete(m.cache, id)
 }
+
+func (m *manager) Updates() <-chan Update {
+	updates := m.updates.Load().(chan Update)
```

Comment:
Possibly, it does add some unfortunate complexity. Once I start using the Updates channel in the ReadinessManager I could just push its creation to the constructor and get rid of all this logic.